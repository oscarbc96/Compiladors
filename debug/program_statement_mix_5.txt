FLEX: Detected mode PRGM
BISON: Starting in mode PRGM
FLEX: COMMENT -> FLEX: ASSIGN yytex: :=
FLEX: INT yytex: 1 -> 1
FLEX: FOR yytex: for
BISON: Checking mode
FLEX: PARENTHESIS_OPEN yytex: (
BISON: Looking for ID x
FLEX: IN yytex: in
FLEX: PARENTHESIS_OPEN yytex: (
FLEX: ID yytex: total -> total
BISON: Looking for ID total
FLEX: ADD yytex: +
FLEX: INT yytex: 1 -> 1
FLEX: PARENTHESIS_CLOSE yytex: )
FLEX: RANGE yytex: ..
FLEX: PARENTHESIS_OPEN yytex: (
FLEX: ID yytex: total -> total
BISON: Looking for ID total
FLEX: ADD yytex: +
FLEX: ID yytex: total -> total
BISON: Looking for ID total
FLEX: MULTIPLY yytex: *
FLEX: ID yytex: total -> total
BISON: Looking for ID total
FLEX: PARENTHESIS_CLOSE yytex: )
FLEX: PARENTHESIS_CLOSE yytex: )
FLEX: DO yytex: do
FLEX: ID yytex: total -> total
FLEX: ASSIGN yytex: :=
FLEX: INT yytex: 1 -> 1
FLEX: IF yytex: if
BISON: Checking mode
FLEX: PARENTHESIS_OPEN yytex: (
FLEX: ID yytex: total -> total
BISON: Looking for ID total
FLEX: COMP yytex: < -> <
FLEX: INT yytex: 10 -> 10
FLEX: OR yytex: or
FLEX: ID yytex: total -> total
BISON: Looking for ID total
FLEX: COMP yytex: = -> =
FLEX: INT yytex: 5 -> 5
FLEX: AND yytex: and
FLEX: ID yytex: total -> total
BISON: Looking for ID total
FLEX: COMP yytex: = -> =
FLEX: ID yytex: total -> total
BISON: Looking for ID total
FLEX: PARENTHESIS_CLOSE yytex: )
FLEX: THEN yytex: then
FLEX: ASSIGN yytex: :=
FLEX: INT yytex: 3 -> 3
FLEX: MULTIPLY yytex: *
FLEX: INT yytex: 2 -> 2
FLEX: ASSIGN yytex: :=
FLEX: ID yytex: x -> x
FLEX: DIVIDE yytex: /
BISON: Looking for ID x
FLEX: INT yytex: 2 -> 2
FLEX: ADD yytex: +
FLEX: ID yytex: x -> x
BISON: Looking for ID x
FLEX: MULTIPLY yytex: *
FLEX: ID yytex: x -> x
BISON: Looking for ID x
FLEX: SUBSTRACT yytex: -
FLEX: INT yytex: 3 -> 3
FLEX: POW yytex: **
FLEX: INT yytex: 2 -> 2
FLEX: ASSIGN yytex: :=
FLEX: ID yytex: x -> x
FLEX: ADD yytex: +
BISON: Looking for ID x
FLEX: FLOAT yytex: 2.3 -> 2.300000
FLEX: ELSIF yytex: elsif
FLEX: PARENTHESIS_OPEN yytex: (
FLEX: ID yytex: x -> x
BISON: Looking for ID x
FLEX: COMP yytex: > -> >
FLEX: INT yytex: 3 -> 3
FLEX: PARENTHESIS_CLOSE yytex: )
FLEX: THEN yytex: then
FLEX: ID yytex: x -> x
FLEX: SUBSTRACT yytex: -
BISON: Looking for ID x
FLEX: INT yytex: 3 -> 3
FLEX: ELSE yytex: else
FLEX: INT yytex: 4 -> 4
FLEX: ADD yytex: +
FLEX: INT yytex: 2 -> 2
FLEX: FI yytex: fi
FLEX: DONE yytex: done
The compilation has been successful
