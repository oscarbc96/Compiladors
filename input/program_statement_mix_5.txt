calc off

// nested sentences

total := 1

for(x in (total+1)..(total+total*total)) do
  total := 1
  if (total < 10 or total = 5 and total = total) then
    x := 3*2
    y := x / 2 + x * x - 3 ** 2
    z := x + 2.3
  elsif (x > 3) then
    x - 3
  else
    4+2
  fi
done
