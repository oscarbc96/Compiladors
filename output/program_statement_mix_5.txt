1: $t01 := 1
2: total := $t01
3: $t02 := total ADDI 1
4: $t03 := total MULI total
5: $t04 := total ADDI $t03
6: x := $t02
7: IF x LEI $t04 GOTO 9
8: GOTO 42
9: $t05 := 1
10: total := $t05
11: IF total LTI 10 GOTO 17
12: GOTO 13
13: IF total EQ 5 GOTO 15
14: GOTO 31
15: IF total EQ total GOTO 17
16: GOTO 31
17: $t06 := 3 MULI 2
18: x := $t06
19: $t07 := x DIVI 2
20: $t08 := x MULI x
21: $t09 := $t07 ADDI $t08
22: PARAM 3
23: PARAM 2
24: $t10 := CALL POW,2
25: $t11 := $t09 SUBI $t10
26: y := $t11
27: $t12 := I2F x
28: $t13 := $t12 ADDF 2.300000
29: z := $t13
30: GOTO 40
31: IF x GTI 3 GOTO 33
32: GOTO 37
33: $t14 := x SUBI 3
34: PARAM $t14
35: CALL PUTI,1
36: GOTO 40
37: $t15 := 4 ADDI 2
38: PARAM $t15
39: CALL PUTI,1
40: x := x ADDI 1
41: GOTO 7
42: HALT
