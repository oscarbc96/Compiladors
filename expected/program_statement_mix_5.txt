1: $t01 := 1
2: total := $t01
3: $t02 := total ADDI 1
4: $t03 := total MULI total
5: $t04 := total ADDI $t03
6: x := $t02
7: IF x LEI $t04 GOTO 9
8: GOTO 41
9: $t05 := 1
10: total := $t05
11: IF total LTI 10 GOTO 17
12: GOTO 13
13: IF total EQ 5 GOTO 15
14: GOTO 30
15: IF total EQ total GOTO 17
16: GOTO 30
17: $t06 := 3 MULI 2
18: x := $t06
19: $t07 := x DIVI 2
20: $t08 := x MULI x
21: $t09 := $t07 ADDI $t08
22: PARAM 3
23: PARAM 2
24: $t10 := CALL POW,2
25: $t11 := $t09 SUBI $t10
26: y := $t11
27: $t12 := x ADDF 2.300000
28: z := $t12
29: GOTO 39
30: IF x GTI 3 GOTO 32
31: GOTO 36
32: $t13 := x SUBI 3
33: PARAM $t13
34: CALL PUTI,1
35: GOTO 39
36: $t14 := 4 ADDI 2
37: PARAM $t14
38: CALL PUTI,1
39: x := x ADDI 1
40: GOTO 7
41: HALT
